apps/aecore/lib/aecore/channel/channel_off_chain_tx.ex

```diff
@@ -0,0 +1,249 @@
+    false
+  end
+
+  def verify_signature_for_key(
+        %ChannelOffChainTx{signatures: {sig1, sig2}} = state,
+        pubkey
+      ) do
+    binary_form = rlp_encode(state)
+
+    Keys.verify(binary_form, sig1, pubkey) or
+      verify_signature_for_key(%ChannelOffChainTx{state | signatures: {sig2, <<>>}}, pubkey)
+  end
+
+  @spec signature_for_offchain_tx(ChannelOffChainTx.t(), Keys.sign_priv_key()) :: binary()
+  defp signature_for_offchain_tx(%ChannelOffChainTx{} = offchain_tx, priv_key)
+       when is_binary(priv_key) do
+    offchain_tx
+    |> rlp_encode()
+    |> Keys.sign(priv_key)
+  end
+
+  @doc """
+  Signs the offchain transaction with the provided private key.
+  """
+  @spec sign(ChannelOffChainTx.t(), Keys.sign_priv_key()) :: ChannelOffChainTx.t()
+  def sign(%ChannelOffChainTx{signatures: {<<>>, <<>>}} = offchain_tx, priv_key) do
+    sig = signature_for_offchain_tx(offchain_tx, priv_key)
+
+    {:ok, %ChannelOffChainTx{offchain_tx | signatures: {sig, <<>>}}}
+  end
+
+  def sign(%ChannelOffChainTx{signatures: {sig1, <<>>}} = offchain_tx, priv_key) do
+    sig2 = signature_for_offchain_tx(offchain_tx, priv_key)
+
+    if sig2 > sig1 do
+      {:ok, %ChannelOffChainTx{offchain_tx | signatures: {sig1, sig2}}}
+    else
+      {:ok, %ChannelOffChainTx{offchain_tx | signatures: {sig2, sig1}}}
+    end
+  end
+
+  @doc """
+  Creates a new offchain transaction containing a transfer update between the specified accounts. The resulting offchain transaction is not tied to any offchain chainstate.
+  """
+  @spec initialize_transfer(binary(), Keys.pubkey(), Keys.pubkey(), non_neg_integer()) ::
+          ChannelOffChainTx.t()
+  def initialize_transfer(channel_id, from, to, amount) do
+    %ChannelOffChainTx{
+      channel_id: channel_id,
+      updates: [ChannelTransferUpdate.new(from, to, amount)],
+      signatures: {<<>>, <<>>}
+    }
+  end
+
+  @spec offchain_updates(ChannelOffChainTx.t()) :: list(ChannelUpdates.update_types())
+  def offchain_updates(%ChannelOffChainTx{updates: updates}) do
+    updates
+  end
+
+  @doc """
+  Encodes the offchain transaction to a form embeddable in ChannelSoloCloseTx, ChannelSlashTx, ChannelSnapshotTx
+  """
+  @spec encode_to_payload(ChannelOffChainTx.t() | :empty) :: binary()
+  def encode_to_payload(%ChannelOffChainTx{signatures: {sig1, sig2}} = state) do
+    [
+      :binary.encode_unsigned(@signed_tx_tag),
+      :binary.encode_unsigned(@version),
+      [sig1, sig2],
+      rlp_encode(state)
+    ]
+    |> ExRLP.encode()
+  end
+
+  def encode_to_payload(:empty) do
+    <<>>
+  end
+
+  @doc """
+  Decodes the embedded payload of ChannelSoloCloseTx, ChannelSlashTx, ChannelSnapshotTx
+  """
+  @spec decode_from_payload(binary()) :: {:ok, ChannelOffChainTx.t()} | :empty | error()
+  def decode_from_payload(<<>>) do
+    {:ok, :empty}
+  end
+
+  def decode_from_payload([@signed_tx_tag, @version, [sig1, sig2], encoded_tx]) do
```

Comment:
This has to be reconsidered. There is no function currently to deserialize and serialize a ChannelOffChainTx with signatures.

Such a function should maybe be part of SignedTx as a ChannelOffChainTx with signatures is an ChannelOffChainTx inside SignedTx.

Please make sure it works for following binary: 
```Bits.decode58("G6DYbwbT9jsa4M7dKjQAw1bSwcPN3TcaCKtxBBAk28QaXfEugae4HGVNxSHSrSd59vPCLAssWmX8vN48LWwTypFq2FYy8AVAev68Q5WerLhcsCViNAtgNdzdeGhRMYHBHxYwnPdMRjKThcQAogdqjrtnMKnwqVE6awwDPYPmRjsDT3JW8HY8m43SWB9QeiauG3EZx8KnjejyXuNZNPGVjFiGnjQCAhwVb5nPXuhs4KXgTKjB5w1Huch5wJg3HUzyQXhuoCk4gn7vcQvjX7P4rthpo1CMNfJj6WXFowKpVzUCQrVrTYUMHzvbYuymzh1CdcmfgJouxHQmDSHXJwRW6Uw6uwRY192oKMbhsanuLjRJC9YB8SzSLdrSu828c6QLHnWqPedsHwW")```